### PR TITLE
Bezier-rs: Add split for Subpath

### PR DESCRIPTION
**Pull request by hannahli2010**
Implement split for Subpaths
- Returns a single opened subpath if the original was closed
- Returns 2 opened subpaths if the original was open

### CODE REVIEW

**Comment by RobNadal** [comment]
I'm don't think the visualization for the closed split case is as clear as we might want. The original idea behind the gradient was so that all the color shifting along the path was smooth so the point where it abruptly changes recognizable as the new start/end. Because we don't have blending between the segments, I think that is lost. I think we'd be closer to that goal if the start and end colors were more similar, ie. red to orange, because then the intermediate colors would be smoother.

Alternatively, I wonder if we could still apply a gradient on each segment (similar to what you tried for the entire subpath), and specify the gradient start and end on endpoint / startpoint. It wouldn't technically be applied 'properly' in cases were the user manipulates a cubic / quadratic to loop on itself or where the endpoint / startpoints are touching, but it would work nicely in our default setups.

File: libraries/bezier-rs/src/subpath/transform.rs [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+				let target_curve_index = scaled_t.floor() as i32;
+				let mut target_curve_t = scaled_t % 1.;
+				let num_manipulator_groups = self.manipulator_groups.len();
+
+				// The only case where `curve` would be `None` is if the provided argument was 1
+				// But the above if case would catch that, since `target_curve_t` would be 0.
+				let curve = self.iter().nth(target_curve_index as usize);
+				let curve = curve.unwrap_or_else(|| {
+					target_curve_t = 1.;
```
[the remark this answers is not in the document]
Comment (hannahli2010) [reply libraries/bezier-rs/src/subpath/transform.rs:25]: Sorry, I'm not sure I follow with what you are suggesting here. Could you elaborate?

File: libraries/bezier-rs/src/subpath/transform.rs [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+				let number_of_curves = self.len_segments() as f64;
+				let scaled_t = t * number_of_curves;
+
+				let target_curve_index = scaled_t.floor() as i32;
+				let mut target_curve_t = scaled_t % 1.;
+				let num_manipulator_groups = self.manipulator_groups.len();
+
+				// The only case where `curve` would be `None` is if the provided argument was 1
+				// But the above if case would catch that, since `target_curve_t` would be 0.
```
[the remark this answers is not in the document]
Comment (hannahli2010) [reply libraries/bezier-rs/src/subpath/transform.rs:22]: Oops, I was supposed to delete that. Thanks for pointing it out

File: libraries/bezier-rs/src/subpath/transform.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (RobNadal) [reply libraries/bezier-rs/src/subpath/transform.rs:25]: Hmm, I thought using `if let` in the if-statement would let you also use beyond the scope too, but it seems it doesn't. 
Just do something like
```
if curve.is_nome() {
  curve = self.iter().nth(target_curve_index as usize);
  target_curve_t = 1.;
}
```

File: libraries/bezier-rs/src/subpath/transform.rs [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+							out_handle: None,
+						});
+					}
+
+					if !second_split.is_empty() {
+						second_split[0].in_handle = second_bezier.handle_end();
+					}
+
+					// Push a new manipulator group to represent the location that was split at
+					// Add the manipulator group except for the case where the splitting point is in the first anchor point
```
[the remark this answers is not in the document]
Comment (hannahli2010) [reply libraries/bezier-rs/src/subpath/transform.rs:68]: The second line of the comment is slightly different, but I'll make some more changes

**Comment by RobNadal** [comment]
LGTM!

**Review by Keavon — APPROVED** [review]

**Pull request merged**